lib/MongoDB/MongoClient.pm

```diff
@@ -1628,14 +1654,19 @@ sub send_retryable_write_op {
 
     # If we get this far and there is no session, then somethings gone really
     # wrong, so probably not worth worrying about.
-    #
+
     # increment transaction id before write, but otherwise is the same for both attempts
-    $op->session->_server_session->_increment_transaction_id;
+    $op->session->_increment_transaction_id;
```

Comment:
I think we should increment the transaction id here only if not in a transaction -- and for a transaction, when we enter the "starting" state, we should increment the transaction.  I think that simplifies some of the state tracking around _active_transaction.

Reply:
`_increment_transaction_id` already has a guard on it in ClientSession if you're in a transaction? Unless i've missed something here. I've added a comment anyway to say its a no-op if in a transaction

Reply:
ok